src/binary-search-tree.js

```diff
@@ -1,10 +1,77 @@
 // https://msdn.microsoft.com/en-us/library/aa289150(v=vs.71).aspx
+class Queue {
```

Comment:
This class should be imported from its own file instead of being put in another class's file.